data/frame.go

```diff
@@ -47,6 +47,15 @@ type Frame struct {
 	Meta *FrameMeta
 }
 
+func (f *Frame) DeepCopy() *Frame {
```

Comment:
For now, frame does not need to be a root object, but implementing DeepCopy allows an upstream struct to include Frame and use standard k8s codegen tools